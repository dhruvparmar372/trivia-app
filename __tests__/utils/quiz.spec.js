import {
  isQuizComplete,
  getFirstUnansweredQuestion,
  isAnswerCorrect,
  getQuizScore
} from "source/utils/quiz";

function getQuiz(complete = false) {
  return {
    id: 1,
    questions: [
      {
        id: 1,
        type: "boolean",
        question: "1 + 1 = 2?",
        correct_answer: "True",
        incorrect_answers: ["False"],
        ...(complete ? { answer: { value: "True" } } : {})
      }
    ]
  };
}

describe("quiz utils", () => {
  test("return false if quiz is not present", () => {
    expect(isQuizComplete()).toMatchSnapshot();
  });
  test("return false if quiz is not complete", () => {
    expect(isQuizComplete(getQuiz())).toMatchSnapshot();
  });
  test("return true if quiz is complete", () => {
    expect(isQuizComplete(getQuiz(true))).toMatchSnapshot();
  });

  test("return first unanswered question", () => {
    expect(isQuizComplete(getQuiz())).toMatchSnapshot();
  });
  test("return null if no unanswered questions", () => {
    expect(isQuizComplete(getQuiz(true))).toMatchSnapshot();
  });

  test("return true if answer is correct", () => {
    const answeredQuestion = getQuiz(true).questions[0];
    expect(
      isAnswerCorrect(answeredQuestion, answeredQuestion.answer)
    ).toMatchSnapshot();
  });
  test("return false if answer is not correct", () => {
    const answeredQuestion = getQuiz(true).questions[0];
    answeredQuestion.answer.value = "False";
    expect(
      isAnswerCorrect(answeredQuestion, answeredQuestion.answer)
    ).toMatchSnapshot();
  });

  test("return quiz score", () => {
    expect(getQuizScore(getQuiz())).toMatchSnapshot();
  });
});
