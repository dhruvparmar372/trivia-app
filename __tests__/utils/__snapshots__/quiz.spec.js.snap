// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz utils return false if answer is not correct 1`] = `false`;

exports[`quiz utils return false if quiz is not complete 1`] = `false`;

exports[`quiz utils return false if quiz is not present 1`] = `false`;

exports[`quiz utils return first unanswered question 1`] = `false`;

exports[`quiz utils return null if no unanswered questions 1`] = `true`;

exports[`quiz utils return quiz score 1`] = `
Object {
  "correct": 0,
  "total": 1,
}
`;

exports[`quiz utils return true if answer is correct 1`] = `true`;

exports[`quiz utils return true if quiz is complete 1`] = `true`;
